File: FE/src/service/OrderService.js
import api from "../api/api";

export const createOrder = (data, orderId) => {
  return api.post(`/orders/create-order`, {
    ...data,
    orderId: orderId,
  });
};
export const createPaymentVnpay = (amount, orderId) => {
  return api.get(`/orders/create-payment`, {
    params: {
      amount: (amount * 25000).toFixed(0),
      orderId: orderId,
      bankCode: "NCB",
    },
    headers: {
      "Content-Type": "application/x-www-form-urlencoded",
    },
  });
};

export const getListNewOrder = (keySearch, pageNumber , pageSize ) => {
  return api.get(`/orders/new-orders`, {
    params: {
      keySearch: keySearch,
      pageNumber: pageNumber,
      pageSize: pageSize,
    },
  });
};
export const getAllOrders = ( pageNumber , pageSize ) => {
  return api.get(`/orders/all-orders`, {
    params: {
      pageNumber: pageNumber,
      pageSize: pageSize,
    },
  });
};
export const getOrdersByCondition = ( data ) => {
  return api.get(`/orders/orders-condition`, {
    params: {
      pageNumber: data.pageNumber,
      pageSize: data.pageSize,
      status: data.orderStatus,
      orderId: data.orderId,
      startDate: data.startDate,
      endDate: data.endDate,
      sortByDateInc: data.sortByDateInc,
    },
  });
};
export const updateOrderStatus = (orderId, status) => {
  return api.put(`/orders/update-order-status`,null, {
    params: {
      orderId: orderId,
      status: status,
    },
  });
};
export const getOrdersByEmail = (email) => {
  return api.get(`/orders`, {
    params: {
      email: email,
    },
  });
};
export const getOrderById = (orderId ) => {
  return api.get(`/orders/${orderId}` );
};
export const delteOrders = (data ) => {
  return api.delete(`/orders/delete-orders`, {
    data: data
  });
};

export const deleteOrderById = (orderId ) => {
  return api.delete(`/orders/delete-order/${orderId}` );
};

File: BE/src/main/java/com/example/Fstock/service/Service/OrderService.java
package com.example.Fstock.service.Service;

import com.example.Fstock.dto.request.CreateOrderRequest;
import com.example.Fstock.dto.response.OrderDto;
import com.example.Fstock.entity.Order;
import com.example.Fstock.enums.OrderStatus;
import org.springframework.data.domain.Page;

import java.time.LocalDate;
import java.util.List;
import java.util.Set;

public interface OrderService {
    void createOrder (CreateOrderRequest createOrderRequest);
    void updateOrderStatus(String orderId, String status);
    void deleteOrders(List<Integer> orderIds);
    void deleteOrderById(Integer orderId);
    OrderDto getOrderById(Integer orderId);
    Page<OrderDto> getListNewOrders(int pageNumber, int pageSize, int keySearch);
    Page<OrderDto> getAllOrders(int pageNumber, int pageSize);
    Set<OrderDto> getOrdersByEmail(String email);
    Page<OrderDto> getOrderByCondition(int pageNumber, int pageSize, String status, Integer orderId, LocalDate startDate, LocalDate endDate, boolean sortByDateInc);
    }
